source/globalCommands.py

```diff
@@ -1188,6 +1188,36 @@ def script_review_endOfLine(self,gesture):
 	script_review_endOfLine.__doc__=_("Moves the review cursor to the last character of the line where it is situated in the current navigator object and speaks it")
 	script_review_endOfLine.category=SCRCAT_TEXTREVIEW
 
+	def script_review_currentSymbol(self,gesture):
+		info=api.getReviewPosition().copy()
+		info.expand(textInfos.UNIT_CHARACTER)
+		text = info.text
+		curLanguage = None
+		if config.conf['speech']['autoLanguageSwitching']:
+			for field in info.getTextWithFields({}):
+				if isinstance(field,textInfos.FieldCommand) and field.command=="formatChange":
+					curLanguage=field.field.get('language')

```

Comment:
Please extract this (lines 1195 -1201 inclusive) to a helper function `_getCurrentLanguageForTextInfo(info)`. This improves the readability for this script by putting the focus on the higher level algorithm.

``` Python
def _getCurrentLanguageForTextInfo(self, info):
	curLanguage = None
	if config.conf['speech']['autoLanguageSwitching']:
		for field in info.getTextWithFields({}):
			if isinstance(field, textInfos.FieldCommand) and field.command == "formatChange":
				curLanguage = field.field.get('language')
	if curLanguage is None:
		curLanguage = speech.getCurrentLanguage()
	return curLanguage
```